pkg/generic/meta.go

```diff
@@ -0,0 +1,60 @@
+package generic
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/label"
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type OwnerClusterGetter func(metav1.ObjectMetaAccessor) (capi.Cluster, bool, error)
+
+func TryGetClusterName(object metav1.ObjectMetaAccessor) (string, bool) {
```

Comment:
I'll add code docs here

Reply:
Since cluster name can't be empty, what about only returning `string` and make consumers check `if string != ""` instead of checking the returned bool value?

Reply:
This felt more explicit, like trying to get env var with `os.LookupEnv("var_name")